man/man8/zfs.8

```diff
@@ -2585,6 +2585,24 @@ Any property specified on the command line using the
 .Fl o
 option is ignored.
 If the target filesystem already exists, the operation completes successfully.
+.It Fl n
+Do a dry-run
+.Pq Qq No-op
+creation.
+No datasets will be created.
+This is useful in conjunction with the
+.Fl v
+or
+.Fl P
+flags to validate properties that are passed via
+.Fl o
+options and those implied by other options.
+The actual dataset creation can still fail due to insufficient privileges or
+available capacity.
+.It Fl P
+Print machine-parseable verbose information about the created dataset.
+.It Fl v
+Print verbose information about the created dataset.
```

Comment:
Can we be more specific about what information will be printed?  E.g. would it be accurate to say that all locally-set properties will be listed, including those set automatically (e.g. refreservation for volumes)?  If we can't document the output specifically, it's hard to consume it programmatically since we could change the output in the next release.

Reply:
While this is a departure from what is done with other parsable options, it is the right thing to do.

```
     zfs create [-Pnpv] [-o property=value]... filesystem
...
       -P  Print machine-parsable verbose information about the created
           dataset.  Each line of output contains a key and one or two values,
           all separated by tabs.  The create_ancestors and create keys have
           filesystem as their only value.  The create_ancestors key only
           appears if the -p option is used.  The property key has two values,
           a property name that property's value.  The property key may appear
           zero or more times, once for each property that will be set local
           to filesystem due to the use of the -o option.
...
     zfs create [-ps] [-b blocksize] [-o property=value]... -V size volume
...
       -P  Print machine-parsable verbose information about the created
           dataset.  Each line of output contains a key and one or two values,
           all separated by tabs.  The create_ancestors and create keys have
           volume as their only value.  The create_ancestors key only appears
           if the -p option is used.  The property key has two values, a prop‐
           erty name that property's value.  The property key may appear zero
           or more times, once for each property that will be set local to
           volume due to the use of the -b or -o options, as well as
           refreservation if the volume is not sparse.
```